src/main/java/bio/terra/service/dataset/DatasetTableDao.java

```diff
@@ -230,13 +275,17 @@ List<Column> retrieveColumns(Table table) {
     return jdbcTemplate.query(
         sqlSelectColumn,
         new MapSqlParameterSource().addValue("table_id", table.getId()),
-        (rs, rowNum) ->
-            new Column()
-                .id(rs.getObject("id", UUID.class))
-                .table(table)
-                .name(rs.getString("name"))
-                .type(TableDataType.fromValue(rs.getString("type")))
-                .arrayOf(rs.getBoolean("array_of"))
-                .required(rs.getBoolean("required")));
+        getColumnRowMapper(table));
+  }
+
+  private RowMapper<Column> getColumnRowMapper(Table table) {
```

Comment:
Is this the same as the one in `SnapshotTableDao`? It'd be nice to share code if possible.

Reply:
Yeah, I moved this to `DaoUtils`